resources/osl_mysql_test.rb

```diff
@@ -10,16 +10,35 @@
 property :server_password, String, default: 'osl_mysql_test'
 property :encoding, String, default: 'utf8mb4'
 property :collation, String, default: 'utf8mb4_unicode_ci'
+property :version, [String, nil]
 property :database_parameters, Hash, default: {}
 
 # Install the mariadb package, set up the service, set up the user, then set up the given database
 action :create do
-  # Install the package, and set up the service
-  mariadb_server_install 'osl-mysql-test' do
-    password new_resource.server_password
-    setup_repo false
-    action [:install, :create]
+  # Check to see if we are running on CentOS, we will be forcing a version if none was given.
+  if platform?('centos') && Gem::Version.create(node['platform_version']) < Gem::Version.create(8) && !new_resource.version
+    new_resource.version = '10.11'
   end
+  # Decide if we should install from the MariaDB respository in the event the stock distro has an outdated version
+  if new_resource.version || node['osl-mysql']['test_mariadb_repo']
```

Comment:
Why is this attribute being used here?

Reply:
This was an earlier attempt to remove the need to specify the `version` tag for every instance this resource is used, only requiring the first resource to need it. I've just pushed an updated version which should not encounter any errors.